crates/connector_proxy/src/apis.rs

```diff
@@ -38,30 +38,68 @@ impl FlowOperation for FlowMaterializeOperation {}
 // InterceptorStream defines the type of input and output streams handled by interceptors.
 pub type InterceptorStream = Pin<Box<dyn Stream<Item = std::io::Result<Bytes>> + Send + Sync>>;
 
-// The generic param "T" below is bounded by FlowOperation.
-// A converter is a function that contains the specific stream-handling logic of an interceptor.
-type ConverterFn<T> = Box<dyn Fn(&T, InterceptorStream) -> Result<InterceptorStream, Error>>;
-// An intercept is characterized by a pair of converters, corresponding to the handling logic of request and response streams, respectively.
-pub type RequestResponseConverterPair<T> = (ConverterFn<T>, ConverterFn<T>);
 pub trait Interceptor<T: FlowOperation> {
-    fn get_converters() -> RequestResponseConverterPair<T> {
```

Comment:
Now the converters are part of the interceptor function calls.
The reason is that - 
1. request and response now have different APIs. The request converter takes in an additional connector PID. 
2. Some shared variables needs to be kept to pass from request to response(e.g. the bindings from [validate request](https://github.com/estuary/flow/blob/master/go/capture/driver/airbyte/driver.go#L297)). And these are easier to be maintained via `Arc<Mutex>` variables in the interceptor object.